python/src/HSICEstimatorImplementation_doc.i.in

```diff
@@ -0,0 +1,360 @@
+%define OT_HSICEstimator_doc
+"Base class of HSICEstimator.
+
+Available constructors:
+    HSICEstimatorImplementation()
+    HSICEstimatorImplementation(covarianceList, X, Y, estimatorType)
+    HSICEstimatorImplementation(covarianceList, X, Y, estimatorType, weightFunction)
```

Comment:
There are several issues here as can be seen on the [compiled HTML file](https://184-159671877-gh.circle-artifacts.com/0/html/user_manual/_generated/openturns.HSICEstimatorGlobalSensitivity.html#openturns.HSICEstimatorGlobalSensitivity)

1. The formatting is off. An example of correct formatting can be found [here](https://github.com/openturns/openturns/blob/b5797d7e4a71c71faf86df51f26ad0d8d551ad08/python/src/SparseMethod_doc.i.in).
2. The "Parameters" section is missing.
3. `HSICEstimatorImplementation` must not be mentioned in the API doc, because Implementation classes never are. These constructors actually appear on the `HSICEstimator` (i.e. the interface class) doc page, so the constructors must refer to `HSICEstimator` instead of `HSICEstimatorImplementation`.

Note that 1. and 2. also hold for the other API doc pages in this PR.

Reply:
OK, thanks for the reminder.